### PR TITLE
Update misc tools

### PR DESCRIPTION
**Pull request by jacob-delgado**

### CODE REVIEW

**Review by dhawton — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]

File: docker/build-tools/Dockerfile [hunk 1]
```diff
@@ -52,7 +52,7 @@ ENV GOLANG_PROTOBUF_VERSION=v1.33.0
 ENV GO_BINDATA_VERSION=v3.1.2
 ENV GO_JUNIT_REPORT_VERSION=df0ed838addb0fa189c4d76ad4657f6007a5811c
 ENV HADOLINT_VERSION=v2.12.0
-ENV HELM3_VERSION=v3.14.2
+ENV HELM3_VERSION=v3.14.3
 ENV HUGO_VERSION=0.123.8
```
Comment (ericvn) [inline docker/build-tools/Dockerfile:56]: Not sure if we wait for a .1, but
```suggestion
ENV HUGO_VERSION=0.124.0
```

**Comment by ericvn** [comment]
Kubernetes 1.29.3 is live as well.

File: docker/build-tools/Dockerfile [hunk 2]
```diff
@@ -481,7 +481,7 @@ FROM ubuntu:jammy as base_os_context
 
 ENV DEBIAN_FRONTEND=noninteractive
 
-ENV CONTAINERD_VERSION=1.6.28-1
+ENV CONTAINERD_VERSION=1.6.28-2
 ENV DOCKER_VERSION=5:25.0.4-1~ubuntu.22.04~jammy
 ENV DOCKER_BUILDX_VERSION=0.13.0-1~ubuntu.22.04~jammy
```
Comment (ericvn) [inline docker/build-tools/Dockerfile:486]:
```suggestion
ENV DOCKER_VERSION=5:25.0.5-1~ubuntu.22.04~jammy
ENV DOCKER_BUILDX_VERSION=0.13.1-1~ubuntu.22.04~jammy
```

**Comment by jacob-delgado** [comment]
cc @kyessenov looks like there is a problem with gcc9 in the https://github.com/istio/tools/blob/master/docker/build-tools/proxy-tsan-instrumented-libcxx.sh for our current builds.

Are you able to look into this? This would happen on any update and isn't limited to updating any tools.

**Comment by kyessenov** [comment]
Envoy is moving towards C++20, so we probably need to bump all toolchains. We're waiting for Android release to fully declare C++20.

**Comment by kyessenov** [comment]
@jacob-delgado we're in a in-between place - since we also need clang 16. let's wait for Envoy to fully upgrade toolchains and then pull updates from their release docker. I'm not sure whether this will break VM builds again like last time.

**Pull request merged**